std/typecons.d

```diff
@@ -149,25 +157,37 @@ public:
     ~this()
     {
         debug(Unique) writeln("Unique destructor of ", (_p is null)? null: _p);
-        if (_p !is null) delete _p;
-        _p = null;
+        release();
     }
-    /** Returns whether the resource exists. */
-    @property bool isEmpty() const
-    {
-        return _p is null;
-    }
-    /** Transfer ownership to a $(D Unique) rvalue. Nullifies the current contents. */
-    Unique release()
+
+    /** Frees the underlying $(D RefT) and nulls it */
+    void release()
     {
         debug(Unique) writeln("Release");
-        auto u = Unique(_p);
-        assert(_p is null);
-        debug(Unique) writeln("return from Release");
-        return u;
+        if (_p !is null)
+        {
+            import core.memory : GC;
+
+            destroy(_p);
+            GC.free(cast(void*)_p);
```

Comment:
If you look at the commit history of this PR, you will see that I originally tried using `malloc` and `free`, but things fell apart when it came to non-static, nested classes and structs (ones that required a closure). `emplace`, in its current form, just doesn't work for them. See https://issues.dlang.org/show_bug.cgi?id=14402 and the comments I've already made on this topic [above](#issuecomment-89867749).

If I'm wrong and you _can_ initialize types requiring a closure using something besides `new`, please let me know, but I couldn't figure out any way to get them properly initialized with the frame pointer of their context.